src/gui/org/deidentifier/arx/gui/view/impl/menu/DialogOrderSelection.java

```diff
@@ -303,19 +303,42 @@ protected void createButtonsForButtonBar(final Composite parent) {
             public void widgetSelected(final SelectionEvent e) {
                 String file = controller.actionShowOpenFileDialog(getShell(), "*.csv"); //$NON-NLS-1$
                 if (file != null){
-                    String[] array = loadArray(file, Charset.defaultCharset());
-                    if (array != null) {
-                        
-                        // Select string
-                        for (int i=0; i<combo.getItems().length; i++){
-                            if (combo.getItem(i).equals("String")) { //$NON-NLS-1$
-                                combo.select(i);
+                    ArrayList<String> fileData = loadFile(file, Charset.defaultCharset());
+                    long fileDataLines = fileData.size();
+
+                    if (fileDataLines != 0) {
+                        // remove values that are not present in the attribute's domain
+                        fileData.retainAll(Arrays.asList(elements));
+
+                        if (fileData.isEmpty() || fileData.size() != elements.length) {
+                            // import failed, file contains no or not all of the
+                            // attribute's domain
+                            controller.actionShowInfoDialog(getShell(),
+                                                               Resources.getMessage("DialogOrderSelection.16"),
+                                                               Resources.getMessage("DialogOrderSelection.17"));
+                        } else if (fileDataLines != fileData.size()) {

```

Comment:
This does not solve the problem very wel. First, it is akward that this condition is checked *after* values have been retained. This is very difficult to read, because the value of fileDataLines has been set *before* values have been returned. This should be the first check which is performed before values are even retained. However, I have another suggestion. Please throw an Exception in the method loadFile(...) as soon as more than elements.size() lines have been read. This immediately stops the loading of data, when it becomes obvious that something is wrong. This check can then be removed.